source/opt/aggressive_dead_code_elim_pass.cpp

```diff
@@ -164,8 +163,7 @@ bool AggressiveDCEPass::AllExtensionsSupported() const {
 bool AggressiveDCEPass::IsDead(Instruction* inst) {
   if (IsLive(inst)) return false;
   if ((inst->IsBranch() || inst->opcode() == SpvOpUnreachable) &&
-      !IsStructuredHeader(context()->get_instr_block(inst), nullptr, nullptr,
-                          nullptr))
+      context()->get_instr_block(inst)->GetMergeInst() == nullptr)
```

Comment:
I'm trying to understand how this is the same as the existing code.  The old `IsStructuredHeader` seemed to do a lot of checks that are now seemingly gone.

Reply:
The `IsStructuredHeader` essentially does the same thing.  It first checks that the given block is not nullptr.  When inlined, that check is not needed because `inst` is  known to be either a branch or `OpUnreachable`. It must have a basic block.

Then it gets the merge instruction as the new code does.  If it is null then it returns false.  If not it extracts information from the merge instruction and store it to the three output parameters if the pointers to them are not null.  In this case, this is skipped because the pointers to the three output parameters are all nullptr.  Then it returns true.

Reply:
Thanks.  That makes sense.  LGTM.